src/utils/utils/units.ts

```
@@ -63,3 +52,138 @@ export const getHumanizedSize = (size: string, bytesOption: 'withB' | 'withoutB'
+  // value CAN BE a floating point number
+  const { unit, value } = getHumanizedSize(quantityString, 'withoutB');
+
+  return { unit, value };
+};
+
+export const quantityToString = ({ unit, value }: Quantity) =>
+  `${value}${unit === 'B' ? '' : unit}`;
+
+/**
+ * Converts bytes to a larger unit (if the corresponding value for that unit is >= 1)
+ * @param value - The value to convert.
+ * @param base - The base to convert to.
+ * @returns The converted value and unit.
+ */
+const convertBytes = (value: number, base: 'BINARY' | 'DECIMAL') => {
+  if (value === 0) {
+    return { unitIndex: 0, value };
+  }
+
+  const divisor = base === 'BINARY' ? 1024 : 1000;
+  let index = 0;
+
+  while (value % divisor === 0) {
+    value /= divisor;
+    index++;
+  }
+
+  return { unitIndex: index, value };
+};
+
+/**
+ * Formats the quantity string that is in bytes format to a more simplified quantity string in integer + unit format (if the conversion to an integer is possible).
+ * e.g. '1024' to 1Ki or '1000000' to 1M
+ *
+ * @param quantityString string to format, must be in a {@link https://github.com/kubevirt/kubevirt/blob/205d9455db56d5fcd42fb331122cfef358f19a69/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L33 Kubernetes Quantity} format
+ * @returns formatted quantity string
+ */
+export const formatQuantityString = (quantityString: string) => {
+  if (isEmpty(quantityString)) {
+    return null;
+  }
+
+  const unit = extractUnitFromQuantityString(quantityString);
+
+  // special case when unit = 'm' can occur, e.g. if we create a PVC with floating point number like 0.2Gi (1/5 Gi) which can't be converted to bytes (power of 2 is not divisible by 5), 'm' then stands for milibytes
+  if (isQuantityUnit(unit) || unit === 'm') {
+    return quantityString;
+  }
+
+  if (!isEmpty(unit)) {
+    // ERROR: formatQuantityString helper called with quantityString argument which is not in Quantity format
+    return quantityString;
```

I'm not sure how we should proceed in case of error (covers also exotic units above). Parsing routines have different strategies is such cases (throwing exception, default values, empty values, returning what's already parsed, etc). 
The components that use this utility have no clear fallback in case of error so throwing exception seems a bad choice.
What do you think about returning some safe but invalid value instead? i.e. zero?
For the code that would need to detect an error we could also provide a `isQuantityString` helper.

---

Returning a zero seems reasonable. Although ideally we should raise an error before calling `formatQuantityString`, so I like the idea of checking the value with `isQuantityString` helper first.

I briefly looked at the usages of `formatQuantityString`, and it is mostly checking pvc size, which comes from useK8sWatchResource so the PVC's yaml should already be formatted correctly by backend.

Can we do the `isQuantityString` helper in a followup? I don't think we have any bugs raised regarding reading invalid values from pvc's yaml right now, so it seems like a nice-to-have, but not a must-have I would say.